k2/csrc/properties.cc

```diff
@@ -10,12 +11,23 @@
 #include <algorithm>
 #include <unordered_set>
 #include <vector>
+#include <stack>
 
 #include "k2/csrc/fsa.h"
 #include "k2/csrc/fsa_algo.h"
 
 namespace k2 {
 
+static constexpr int8_t kNotVisited = 0;  // a node that has not been visited
+static constexpr int8_t kVisiting = 1;    // a node that is under visiting
+static constexpr int8_t kVisited = 2;     // a node that has been visited
+// depth first search state
+struct DfsState {
+  int32_t state;      // state number of the visiting node
+  int32_t arc_begin;  // arc index of the visiting arc
+  int32_t arc_end;    // end of the arc index of the visiting node
+};
```

Comment:
If these struct and constexpr is useful in different files, I'm OK with it in namespace `k2`, but a better place is in a `header` file, such as `utils.h`.

As `fsa_algo.cc` defines same thing, we definitely should avoid such duplicate code.